Extension/src/LanguageServer/client.ts

```diff
@@ -2757,10 +2757,38 @@ export class DefaultClient implements Client {
         }
     }
 
-    private logTelemetry(notificationBody: TelemetryPayload): void {
+    private excessiveFilesWarningShown: boolean = false;
+    private async logTelemetry(notificationBody: TelemetryPayload): Promise<void> {
         if (notificationBody.event === "includeSquiggles" && this.configurationProvider && notificationBody.properties) {
             notificationBody.properties["providerId"] = this.configurationProvider;
         }
+
+        const showExcessiveFilesWarning = new PersistentWorkspaceState<boolean>('CPP.showExcessiveFilesWarning', true);
+        if (!this.excessiveFilesWarningShown && notificationBody.event === 'ParsingStats' && showExcessiveFilesWarning.Value) {
```

Comment:
Can you move this before the ParsingStats string string check so it can potentially skip that?

Reply:
You're talking about the `showExcessiveFilesWarning.Value`?

Reply:
Yes.

Reply:
Sure. This check grew a bit organically and I didn't think about it.